plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/jdt/ui/wizards/NewInterfaceWizardPage.java

```diff
@@ -0,0 +1,86 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Red Hat, Inc and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Red Hat, Inc - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.reddeer.eclipse.jdt.ui.wizards;
+
+import org.eclipse.reddeer.core.reference.ReferencedComposite;
+import org.eclipse.reddeer.jface.wizard.WizardPage;
+import org.eclipse.reddeer.swt.impl.text.LabeledText;
+
+/**
+ * Wizard page for creating a java interface.
+ */
+public class NewInterfaceWizardPage extends WizardPage {
```

Comment:
Basic features are implemented properly, but I am missing another option that wizard dialog offers, how about choosing enclosing type check box and its input box? Also, modifiers and option to choose  adding the extended interfaces? I would consider adding those functionality as well and properly extend test class for this new feature. Thanks!

Reply:
Added support for toggling enclosing type check box and changing its input box:

- toggleEnclosingTypeCheckBox()
- getEnclosingType()
- setEnclosingType()

Working with the extended interfaces will be added soon.

Reply:
UPDATE: I have added functions addExtendedInterfaces() and removeExtendedInterfaces().

Reply:
Added.